packages/react/src/FormattingToolbar/components/DefaultDropdowns/BlockTypeDropdown.tsx

```diff
@@ -15,114 +10,112 @@ import {
+  },
+  {
+    name: "Bullet List",
+    type: "bulletListItem",
+    icon: RiListUnordered,
+  },
+  {
+    name: "Numbered List",
+    type: "numberedListItem",
+    icon: RiListOrdered,
+  },
+];
 
 export const BlockTypeDropdown = <BSchema extends BlockSchema>(props: {
   editor: BlockNoteEditor<BSchema>;
+  items?: BlockTypeDropdownItem[];
 }) => {
   const [block, setBlock] = useState(
     props.editor.getTextCursorPosition().block
   );
 
-  const dropdownItems: ToolbarDropdownProps["items"] = useMemo(() => {
-    const items: ToolbarDropdownProps["items"] = [];
-
-    if ("paragraph" in props.editor.schema) {
-      items.push({
-        onClick: () => {
-          props.editor.focus();
-          props.editor.updateBlock(block, {
-            type: "paragraph",
-            props: {},
-          });
-        },
-        text: "Paragraph",
-        icon: RiText,
-        isSelected: block.type === "paragraph",
-      });
-    }
-
-    if (
-      "heading" in props.editor.schema &&
-      "level" in props.editor.schema.heading.propSchema
-    ) {
-      items.push(
-        ...(["1", "2", "3"] as const).map((level) => ({
-          onClick: () => {
-            props.editor.focus();
-            props.editor.updateBlock(block, {
-              type: "heading",
-              props: { level: level },
-            } as PartialBlock<BSchema>);
-          },
-          text: "Heading " + level,
-          icon: headingIcons[level],
-          isSelected: block.type === "heading" && block.props.level === level,
-        }))
-      );
-    }
-
-    if ("bulletListItem" in props.editor.schema) {
-      items.push({
-        onClick: () => {
-          props.editor.focus();
-          props.editor.updateBlock(block, {
-            type: "bulletListItem",
-            props: {},
-          });
-        },
-        text: "Bullet List",
-        icon: RiListUnordered,
-        isSelected: block.type === "bulletListItem",
-      });
-    }
+  const filteredItems: BlockTypeDropdownItem[] = useMemo(() => {
+    return (props.items || defaultBlockTypeDropdownItems).filter((item) => {
+      // Checks if block type exists in the schema
+      if (!(item.type in props.editor.schema)) {
+        return false;
+      }
+
+      // Checks if props for the block type are valid
+      for (const [prop, value] of Object.entries(item.props || {})) {
+        const propSchema = props.editor.schema[item.type].propSchema;
+
+        // Checks if the prop exists for the block type
+        if (!(prop in propSchema)) {
+          return false;
+        }
+
+        // Checks if the prop's value is valid
+        if (
+          propSchema[prop].values !== undefined &&
+          !propSchema[prop].values!.includes(value)
+        ) {
+          return false;
+        }
+      }
+
+      return true;
+    });
+  }, [props.editor, props.items]);
+
+  const shouldShow: boolean = useMemo(
+    () => filteredItems.map((item) => item.type).includes(block.type as string),
```

Comment:
minor detail, you can use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find for this